### DISCUSSION TITLE
Шлем отдельно

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#510] [General]
- Гаронд?
- мертвый Сенграт?
- Горн на ферме?

давай тогда по блокам 

**Паладины Хоринос**

- [x] - Лорд Андре
- [x] - Лорд Хаген
- [ ] - Альбрехт, читает книгу
- [ ] - Ингмар? (мне он нравился в шапочке от легкой брони паладина)
- [x] - Седрик?, тренируется с мечем
- [ ] - Лотар? в верхнем квартале

**ОнД 4 глава Миненталь.** 

Без шлема сидят и ждут, после диалога с ГГ одевают шлем:

- [ ] Горн
- [ ] Сайфер
- [ ] Род
- [ ] Бифф
- [ ] Синтенза
- [ ] Булко

Пример видео с Горном
https://www.youtube.com/watch?v=eq3aTsBaKzI
По  такому же принципу

ставь галочки:-)

************

Ну как то в диалогах с закрытым шлемом герой честно говоря выглядит не очень.

**Comment by dosinabox** [comment]
> Ну как то в диалогах с закрытым шлемом герой честно говоря выглядит не очень.

Зато теперь его можно снимать и выглядит это отлично)

> а на НПС во время диалога Шлем могут одеваться?
> например можно прийти к Горну в 4 главе он будет без шлема. поговорить с ним он оденет шлем и побежит в атаку.

Можно. Но зачем?

**Comment by Kor-Angar** [comment]
я вот думаю как будет выглядит если одержимых(Вино, Фернандо) сделать без капюшона?:-)

**Comment by dosinabox** [comment]
> обновил версию и у помощника опция пропала. 

Вернется в ближайшей тестовой сборке. На релиз заблокировала и не стала включать в мод файлы моделей. В новой версии будет и опция, и все нужные файлы. Вроде как твои тесты прошли успешно.

> т.е. они как бы есть и их как бы нет? :-)

Можешь взять модели здесь: https://worldofplayers.ru/threads/41303

**Comment by dosinabox** [comment]
> я вот думаю как будет выглядит если одержимых(Вино, Фернандо) сделать без капюшона?:-)

Там дурацкие модели - капюшоны не совпадают с лицами. Выглядеть они будут плохо)

**Comment by Kor-Angar** [comment]
> вот стоит Горн без шлема, высматривает орков. вот пошел с героем в бой и одел шлем

визуализировал 
https://www.youtube.com/watch?v=eq3aTsBaKzI

как тебе?


> Возвращаясь к вопросу одевания, снимания шлемаslightly_smiling_face это же реализуется в скриптах так же как и одевание доспехов? slightly_smiling_face

оказывается так же:-)

**Comment by Kor-Angar** [comment]
Лотар в нижнем квартале в шлеме.  в верхнем квартале без шлема :-)
![image](https://user-images.githubusercontent.com/49514764/64985007-9a969a80-d8cc-11e9-945d-96a5eae862d6.png)
![image](https://user-images.githubusercontent.com/49514764/64985011-9e2a2180-d8cc-11e9-8cbe-3b215dd56584.png)

```
instance PAL_203_Lothar(Npc_Default)
{
	name[0] = "Ћотар";
	guild = GIL_PAL;
	id = 203;
	voice = 1;
	flags = NPC_FLAG_IMMORTAL;
	npcType = npctype_main;
	B_SetAttributesToChapter(self,5);
	fight_tactic = FAI_HUMAN_MASTER;
//	EquipItem(self,ItMw_1h_Pal_Sword);
	EquipItem(self,ItMw_1H_Blessed_02);
	B_CreateAmbientInv(self);
	B_SetNpcVisual(self,MALE,"Hum_Head_Psionic",Face_P_NormalBart_Nefarius,BodyTex_P,ITAR_PAL_H);
	Mdl_SetModelFatness(self,1);
	Mdl_ApplyOverlayMds(self,"Humans_Militia.mds");
	B_GiveNpcTalents(self);
	B_SetFightSkills(self,80);
	daily_routine = Rtn_preStart_203;
};


func void Rtn_preStart_203()
{
	TA_Stand_Guarding(8,0,23,0,"NW_CITY_MAINSTREET_01_B");
	TA_Stand_Guarding(23,0,8,0,"NW_CITY_MAINSTREET_01_B");
};

func void Rtn_Start_203()
{
	TA_Stand_Guarding(8,0,23,0,"NW_CITY_LOTHAR");	
	TA_Stand_Guarding(23,0,8,0,"NW_CITY_LOTHAR");	B_SetNpcVisual(self,MALE,"Hum_Head_Psionic",Face_P_NormalBart_Nefarius,BodyTex_P,ITAR_PALN_H);	
};

func void Rtn_Tot_203()
{
	TA_Sleep(8,0,23,0,"TOT");
	TA_Sleep(23,0,8,0,"TOT");
};
```
хотел так же переодеть Горна в Минентале, но фокус не удался

**Comment by Kor-Angar** [comment]
а как сделать броню из тяжелых доспехов паладина и среднего шлема? или нужно новый доспех такой делать?

**Comment by dosinabox** [comment]
> а как сделать броню из тяжелых доспехов паладина и среднего шлема? или нужно новый доспех такой делать?

Нужен отдельный шлем, который будет считаться **оружием**. Или делать новую модель.

https://worldofplayers.ru/threads/41303/post-1039877

**Comment by dosinabox** [comment]
Поскольку модели и скрипты раздельных доспехов и шлемов теперь есть в моде, то можно подумать, кто из паладинов/ОнД не будет носить шлемы:
- Хаген?
- Гаронд?
- мертвый Сенграт?
- Горн в Минентале?
- Горн на ферме?
- ?

**Comment by Kor-Angar** [comment]
> что что-то сломалось и шлемы у этих персонажей пропали из-за какой-то ошибки. Или нет?

не должны. часть же в шлемах будет. в ВК были исключительно положительные отзывы.

*******************

единственное когда шлемы отдельно волосы на голове сквозь шлем пролазят когда герой достаёт оружие. на стриме обращал часто. надо будет отдельно протестить.

*******************

**Comment by dosinabox** [comment]
> единственное когда шлемы отдельно волосы на голове сквозь шлем пролазят когда герой достаёт оружие

Тоже часто вижу, но только у тебя на стримах. DX11 это.

**Comment by dosinabox** [comment]
А включать ли раздельные доспехи и шлемы для ГГ по умолчанию - не знаю...
Вот увидят люди интересное решение с Хагеном и захотят выглядеть так же, а про настройки не знают.

**Comment by dosinabox** [comment]
Что если Хагену/Гаронду сделать особые доспехи без шлемов, с какими-нибудь золотыми вставками или тканями-украшениями?

**Comment by Kor-Angar** [comment]
> А включать ли раздельные доспехи и шлемы для ГГ по умолчанию - не знаю...

не уверен, что стоит. это вопрос к настройки обновления через статую / кресло / ксардаса что то там ещё.

> Что если Хагену/Гаронду сделать особые доспехи без шлемов, с какими-нибудь золотыми вставками или тканями-украшениями?

а в чем фишка? Например там есть офицер стратег короля который находится под командованием Гаронда. возможно он по соц статусу/иерархии выше Гаронда, хотя конкретно на этом задание находится под его командованием. т.е. это я к тому что мы не знаем их иерархии. да Лорд Хаген командующий экспедиции но это не значит что он там самый самый самый. да он Лорд. Паладин короля и так далее. Но и Андре Лорд и Паладин короля и так далее :-) 
у них и так тяжелые доспехи паладина. они члены ордена поэтому по идеи должны как братья иметь одинаковое обмундирование. рядовые в обычных доспехах, офицеры/элита в тяжёлых. 
мы снимаем отдельным НПС шлем что бы подчеркнуть их особенность.

**Comment by Kor-Angar** [comment]
Альбрехт будет в шлеме грызть гранит науки? :-)

**Comment by dosinabox** [comment]
> после диалога с ГГ одевают шлем

Опять же, у игрока сразу возникнет вопрос - "почему они могут ходить без шлемов и надевать их прямо на моих глазах, а я не могу?".

**Comment by Kor-Angar** [comment]
Ты это пишишь в контексте того что бы включить разделение шлемов по дефауту?

**Comment by Kor-Angar** [comment]
Не помню бегал ли я на стриме в доспехах бандита и шапке сеточки паладина по Ярику.  Или в рамках теста пробегал. Вообщем со шлемами и в этом плане тоже надо думать.

**Comment by dosinabox** [comment]
> Вообщем со шлемами и в этом плане тоже надо думать.

А что тут можно придумать? Запрещать носить что-то с чем-то? А как это будет обосновано?

> Ты это пишишь в контексте того что бы включить разделение шлемов по дефауту?

А вот не знаю, как лучше. Наверное, лучше не включать по умолчанию. Но нужно будет хотя бы упомянуть о такой возможности в теме на форуме и в Readme мода.

**Comment by dosinabox** [comment]
> мы снимаем отдельным НПС шлем что бы подчеркнуть их особенность.

Мне кажется, что нужно снять только у Хагена и Гаронда. Может быть, ещё у Андре и у Горна.
Если снимать у всех, кто тренируется или читает, то тогда особенных НПС не останется и вся задумка смазывается)

**Comment by Kor-Angar** [comment]
> А что тут можно придумать? Запрещать носить что-то с чем-то? А как это будет обосновано?

это будет не логично. станет ещё запутаннее. в играх обычно их дают не одновременно. т.е. если броня блатная то ты её собираешь по кусочкам. сначала доспех, потом шлем. тут мы так сходу сделать это не сможем. может быть потом.

> Если снимать у всех, кто тренируется или читает, то тогда особенных НПС не останется и вся задумка смазывается)

и да  и нет, пошел тонкий лёд:-) очень хочется подчеркнуть особенность/эксклюзивность НПС путем снятие шлема, на примере лорда Хагена. Но в ратуши находятся офицеры и становится непонятно почему лорд Хаген без шлема, а Альбрехт читает/пишет целый день в шлеме.. но это уже очень субъективно конечно.
Касательно паладина тренирующегося: он в броне рыцаря, хочется её тоже показать без шапочки :-)

> Мне кажется, что нужно снять только у Хагена и Гаронда. Может быть, ещё у Андре и у Горна.

есть разные ситуации: 
- без шлема
- интерактив с одеванием шлема

что бы окончательно не запутаться я пытаюсь с тобой согласовывать по блокам :-) блок по паладинам в Хориносе мы проработали. 

следующий блок по ОНД в 4 главе с "интерактивом". ввиду того что я не знаю как правильно. я накидываю спектр вариантов. а ты как консерватор выбираешь ;-)

добавил варианты в шапку(первый пост)

**Comment by Kor-Angar** [comment]
> А вот не знаю, как лучше. Наверное, лучше не включать по умолчанию. Но нужно будет хотя бы упомянуть о такой возможности в теме на форуме и в Readme мода.

надо будет сделать видео обзор и рассказать про каждую опцию. а потом его подраскрутить.

**Comment by Kor-Angar** [comment]
у Лорда Андре и  Лорда Хагена, шлемы сняты

**Comment by Kor-Angar** [comment]
такой момент обсуждали ещё раз хотел уточнить:
можно ли сделать доспех из тяжелого паладина + шапка сетка от легко паладина? насколько это трудозатратно?

***********
код такой?
ITAR_PALN_M - Доспехи рыцаря без шлема
ITAR_PALN_H - Доспехи паладина без шлема

ITAR_PAL_M - Доспехи рыцаря с шлемом
ITAR_PAL_H - Доспехи паладина с шлемом

**Comment by Kor-Angar** [comment]
Гаронд в Шапочке :-)
https://www.youtube.com/watch?v=ESIHBUoQqPM&feature=youtu.be

**Comment by dosinabox** [comment]
> можно ли сделать доспех из тяжелого паладина + шапка сетка от легко паладина? насколько это трудозатратно?

Это новый доспех делать и подгонять всё под скелет. Очень трудозатрано. А как вешать шапочки на нпц не знаю. Но наверняка есть способ. Опять же, искать этот способ - трудозатрано.

**Comment by Kor-Angar** [comment]
а старый доспех паладина не разбирается? :)

**Comment by Kor-Angar** [comment]
ОнД 5 глава Хоринос
может в 5 главе Горна  в Хориносе вставить без шлема? :-)

**Comment by Kor-Angar** [comment]
так же в 5 главе в диалоге с Хагеном надо ему наверное шлем одевать.

**Comment by Kor-Angar** [comment]
> а старый доспех паладина не разбирается? :)

![image](https://user-images.githubusercontent.com/49514764/74455551-831d0700-4e96-11ea-8781-d8059ce401b7.png)

странная тенденция :-) но в целом я именно в этом контексте и думал :-)

**Comment by dosinabox** [comment]
В релиз пошли Андрэ и Хаген без шлемов. Посмотрим, что скажет народ)

> а старый доспех паладина не разбирается? :)

Ну, можно сделать новую модель без шлема, но тогда Архол/Инубис будут выглядеть менее защищенными и менее грозными, чем обычные теневые воины. Тогда уже совсем новые доспехи им делать, не паладинские.

**Comment by Kor-Angar** [comment]
Старый доспех без Шлема штука прикольная. Можно подумать как его задействовать 😀Я вот часто думаю насчёт тех доспехов которые стоят на ирдорате как часть интерьера. Почему они такие новые, кто с них пыль стирает. Касательно нежити в доспехах без Шлема, сначалаюо бы на них посмотреть бы 😀

**Comment by Kor-Angar** [comment]
как там Беннет продаёт средние доспехи если опция "разделить доспехи" включена?

**Comment by dosinabox** [comment]
Дает и доспехи, и шлем. Так же с тяжелыми доспехами ОнД и доспехами рыцаря. Доспехи паладина тоже появляются в раздельном виде.

**Comment by Kor-Angar** [comment]
это хорошо

**Comment by Kor-Angar** [comment]
Лорду Андре в версии 24 вернули шлем? :-) Широкая общественность не оценила? :-)

**Comment by dosinabox** [comment]
> В релиз пошли Андрэ и Хаген без шлемов.

Случайно вышло)
Не знаю, рано. Ещё не все приняли Андрэ в доспехах паладина, и тут вдруг - без шлема и он и ещё Хаген.

**Comment by Kor-Angar** [comment]
А в хотфикс поправить можно? 😀

**Comment by Kor-Angar** [comment]
> Ну, можно сделать новую модель без шлема, но тогда Архол/Инубис будут выглядеть менее защищенными и менее грозными, чем обычные теневые воины. Тогда уже совсем новые доспехи им делать, не паладинские.

почему не использовать одну и туже модель но разные текстуру через **visual_skin** ?

```
instance ITAR_PALN_H(C_Item)
{
	name = "Доспехи паладина";
	mainflag = ITEM_KAT_ARMOR;
	flags = 0;
        ...
	wear = WEAR_TORSO;
	visual = "ITAR_PALN_H.3ds";
//	visual_change = "ARMOR_PALN_H.asc";
	visual_change = "Hum_PalH_Armor.asc";
	visual_skin = 1;
	material = MAT_METAL;
	...
};

instance ITAR_PAL_Old(C_Item)
{
	name = "Старые доспехи паладина";
	mainflag = ITEM_KAT_ARMOR;
	flags = 0;
        ...
	wear = WEAR_TORSO;
	visual = "ItAr_Pal_Skel.3ds";
	//visual_change = "ARMOR_PAL_OLD.asc";
	visual_change = "Hum_PalH_Armor.asc";
	visual_skin = 0;
	material = MAT_METAL;
        ...
};
```

**Comment by Kor-Angar** [comment]
а у тебя доспехи Лорда **_Armor_Pal_S.asc_**  имеют одинаковую модель  как и доспех Паладина **_ARMOR_PALN_H.asc_**?

**Comment by dosinabox** [comment]
> а у тебя доспехи Лорда **_Armor_Pal_S.asc_** имеют одинаковую модель как и доспех Паладина **_ARMOR_PALN_H.asc_**?

Нет. У них модель доспеха рудного барона.

> почему не использовать одну и туже модель но разные текстуру через **visual_skin** ?

Потому что в Г2 разработчики забили на этот параметр и не стали именовать текстуры как положено. В результате ни одна текстура доспехов не подходит для смены через **visual_skin**.

**Comment by Kor-Angar** [comment]
> Г2 разработчики забили на этот параметр

жаль что ты [в своих asc](https://worldofplayers.ru/threads/41303/) не использовала правильно наименование текстур -(

вот я поправил в asc название текстур и у меня visual_skin заработал, но не могу понять как текстуру шлема правильно назвать что бы visual_skin заработал и на шлеме в оригинальных asc доспеха паладина.

**Comment by dosinabox** [comment]
Шлемы - не динамические asc/mdm, а статические 3ds/mrm. Скинов у них нет.

> вот я поправил в asc название текстур и у меня visual_skin заработал

Это ты вручную добавил нужную. А если бы разработчики называли подходящие текстуры с приставкой _V0 / _V1, то менять исходники моделей не пришлось бы. Работало бы после изменения одного значения из скриптов.

**Comment by Kor-Angar** [comment]
> А если бы разработчики называли подходящие текстуры с приставкой _V0 / _V1, то менять исходники моделей не пришлось бы. 

по моему у этих текстур правильное название в Г2 НВ :-)
![image](https://user-images.githubusercontent.com/49514764/95791910-0ecb8980-0ceb-11eb-9026-9220b2b776e2.png)

https://steamcommunity.com/sharedfiles/filedetails/?id=2256159459

**Comment by Kor-Angar** [comment]
> а что то у меня ITAR_SLD_S и ITAR_SLD_H стали одинаковые. у них модели одинаковые? 

![image](https://user-images.githubusercontent.com/49514764/95810525-d4c5ac00-0d19-11eb-8281-72b9773f36c9.png)

**Comment by Kor-Angar** [comment]
склеил тяжёлый доспех с шапкой сеточкой
https://vk.com/wall-147438775_4983

**Comment by Kor-Angar** [comment]
а чем эти два файла отличаются? Доспех редактировали?
[G1_Hum_OrgM_Armor.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/5395616/G1_Hum_OrgM_Armor.zip)
[G2_Armor_BDT_H.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/5395617/G2_Armor_BDT_H.zip)

**Comment by Kor-Angar** [comment]
Лёгкий шлем(сеточка) паладина не "налазит на голову" Хагена

**Comment by Kor-Angar** [comment]
> Это новый доспех делать и подгонять всё под скелет. Очень трудозатрано. А как вешать шапочки на нпц не знаю. Но наверняка есть способ. Опять же, искать этот способ - трудозатрано.

получилось сделать через  	on_equip в доспехе. 
```
instance ITAR_MIL_H2(Hum_Mil_H_ARMOR)
{
	name				=	"Тяжелые доспехи ополчения v2";
	visual_skin 			=	2;	
	on_equip				=	Equip_ITAR_PAL_Old;
	on_unequip			=	UnEquip_ITAR_PAL_Old;
};

INSTANCE ITHE_PAL_M2(C_Item)
{
	name = " ольчужный капюшон";
	mainflag = ITEM_KAT_ARMOR;
//	flags =  ITEM_ACTIVE;	// не даёт одеть шлем через  Equip_ITAR_PAL_Old:-)
	flags = ITEM_NFOCUS;
	value = 1000;
	wear = WEAR_Head;
	visual = "ITHE_PAL_M.3ds";
	visual_skin = 0;
	material = MAT_METAL;
	description = name;
};

/******************************************************************************************/
var int EquipITARPALOld;

func void Equip_ITAR_PAL_Old()
{	
		CreateInvItem(self,ITHE_PAL_M2);
		AI_EquipArmor(self,ITHE_PAL_M2);
};
```

**Comment by Kor-Angar** [comment]
> Außerdem habe ich dem Hals der mittleren Drachenjägerrüstung eine Textur zugewiesen, das war beim Original nicht der Fall, weshalb dort stattdessen die graue Default-Textur angezeigt wurde.

https://www.worldofgothic.de/?go=moddb&action=view&fileID=1497&cat=1&page=0&order=0

**Comment by Kor-Angar** [comment]
мы по моему обсуждали вопрос с одеванием / сниманием шлема на скриптах. это актуально?

**Reply by Kor-Angar** [reply]
пока в памяти свежно. накатаю :-) https://www.youtube.com/watch?v=DgtLmMNmYgY

**Reply by Kor-Angar** [reply]
```D
func int C_EquipHelms(var c_npc slf, var int newitm)
{
    var string SlotName; SlotName ="ZS_HELMET";
    
     var c_item itm;  

    itm = Npc_GetSlotItem(slf, SlotName);     // берём надетую шапку
    if Hlp_IsValidItem(itm) // шапка есть
    {
        if (Hlp_IsItem(itm, newitm) == true)
        {
            Hlp_PrintConsole("  НПС экипирован уже нужной шапкой --> выход");
            return false;          
        };
        // снимает шапку      
        Npc_RemoveInvItem(slf, Hlp_GetInstanceID(itm));
        CreateInvItem(slf,Hlp_GetInstanceID(itm));
        //  Npc_RemoveFromSlot(slf, SlotName , 0); // нужно, если НПС управляет герой?
    }; 

    // берём ссылку на новую шапку
    if !Npc_GetInvItem(slf,newitm)
    {
        CreateInvItem(slf,newitm);  // если шапки нет, то создаём
    };   
    
    Npc_PutInSlot(slf,SlotName,item,1);  // визуально одет
    EquipItem(slf,newitm);                       // маркер что предмет одет
    return true;
};
```

**Reply by Kor-Angar** [reply]
![20230318174240_1](https://user-images.githubusercontent.com/49514764/226115235-ca408c77-c9ec-453f-9a64-188162b3cea9.jpg)
